ST_seedDispersal.c

```diff
@@ -0,0 +1,324 @@
+				Species[s]->sd_sgerm = sgerm; //based upon whether we have received/produced seeds that germinated
+			}
+		}
+
+	}
+
+	// calculate whether or not seeds were received/produced this year, this data is used the next time the function is called
+	ForEachSpecies(s)
+	{
+		if (!(Species[s]->use_me && Species[s]->use_dispersal))
+			continue;
+
+		IndivType* indiv;
+
+		// figure out which species in each cell produced seeds...
+		for (i = 0; i < grid_Cells; i++)
+		{
+            row = i / grid_Cols;
+            col = i % grid_Cols;
+            cell = &gridCells[row][col];
+
+            load_cell(row, col);
+
+			cell->mySeedDispersal[s].seeds_present = cell->mySeedDispersal[s].seeds_received =
+					Species[s]->received_prob = 0;
+
+			biomass = 0;	//getting the biggest individual in the species...
+			ForEachIndiv(indiv, Species[s])
+				if (indiv->relsize * Species[s]->mature_biomass
+						> biomass)
+					biomass = indiv->relsize
+							* Species[s]->mature_biomass;
+
+			if (GE(biomass,
+					Species[s]->mature_biomass
+							* Species[s]->sd_Param1))
+			{
+				randomN = RandUni(&grid_rng);
+
+				LYPPT = cell->mySeedDispersal[s].lyppt;
+				float PPTdry = Species[s]->sd_PPTdry, PPTwet =
+						Species[s]->sd_PPTwet;
+				float Pmin = Species[s]->sd_Pmin, Pmax =
+						Species[s]->sd_Pmax;
+
+				//p3 = Pmin, if LYPPT < PPTdry
+				//p3 = 1 - (1-Pmin) * exp(-d * (LYPPT - PPTdry)) with d = - ln((1 - Pmax)/(1 - Pmin)) / (PPTwet - PPTdry), if PPTdry <= LYPPT <= PPTwet
+				//p3 = Pmax, if LYPPT > PPTwet
+
+				presentProb = 0.0;
+				if (PPTdry <= LYPPT && LYPPT <= PPTwet)
+				{
+					float d = -log(((1 - Pmax) / (1 - Pmin)))
+							/ (PPTwet - PPTdry); //log is the natural log in STD c's math.h
+					presentProb = 1 - (1 - Pmin) * exp((-d * (LYPPT - PPTdry)));
+				}
+				else if (LYPPT < PPTdry)
+					presentProb = Pmin;
+				else if (LYPPT > PPTwet)
+					presentProb = Pmax;
+
+				if (LE(randomN, presentProb))
+					cell->mySeedDispersal[s].seeds_present = 1;
+			}
+		}
+
+		// figure out which species in each cell received seeds...
+		for (i = 0; i < grid_Cells; i++)
+		{
+            row = i / grid_Cols;
+            col = i % grid_Cols;
+            cell = &gridCells[row][col];
+
+            load_cell(row, col);
+
+			if (cell->mySeedDispersal[s].seeds_present)
+				continue;
+			receivedProb = 0;
+
+			for (j = 0; j < cell->mySeedDispersal[s].size; j++)
+				if (cell->mySeedDispersal[cell->mySeedDispersal[s].cells[j]].seeds_present)
+					receivedProb += cell->mySeedDispersal[s].dispersalProb[j];
+
+			randomN = RandUni(&grid_rng);
+			if (LE(randomN, receivedProb) && !ZRO(receivedProb))
+				cell->mySeedDispersal[s].seeds_received = 1;
+			else
+				cell->mySeedDispersal[s].seeds_received = 0;
+
+			Species[s]->received_prob = receivedProb;
+		}
+	}
+
+	unload_cell();
+*/
+}
+
+int _do_bulk_dispersal(SppIndex sp){
+    return 0;
+}
+
+void _do_precise_dispersal(int leftoverSeeds, SppIndex sp){
+    return;
```

Comment:
Remove redundant return.

Reply:
@fdpierson I added the explicit return to make it clear to future programmers that this function is intentionally empty until seed dispersal is implemented. I will, however, add a comment as well to make it clearer.